server/src/test/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapperTests.java

```diff
@@ -695,6 +695,42 @@ public void testPointsOnlyFalseWithTermStrategy() throws Exception {
         assertFieldWarnings("tree", "precision", "strategy", "points_only");
     }
 
+    /**
+     *
+     * Test that doc_values parameter correctly parses
+     */
+    public void testDocValues() throws IOException {
+        String trueMapping = Strings.toString(XContentFactory.jsonBuilder().startObject().startObject("type1")
+            .startObject("properties").startObject("location")
+            .field("type", "geo_shape")
+            .field("tree", "quadtree")
+            .field("doc_values", true)
+            .endObject().endObject()
+            .endObject().endObject());
+
+        ElasticsearchParseException e = expectThrows(ElasticsearchParseException.class,
+            () -> createIndex("test").mapperService().documentMapperParser().parse("type1", new CompressedXContent(trueMapping)));
+        assertThat(e.getMessage(), equalTo("geo_shape field [location] indexed using prefix-trees do not support doc_values"));
+
+        // explicit false doc_values
+        String falseMapping = Strings.toString(XContentFactory.jsonBuilder().startObject().startObject("type1")
+            .startObject("properties").startObject("location")
+            .field("type", "geo_shape")
+            .field("tree", "quadtree")
+            .field("doc_values", "false")
+            .endObject().endObject()
+            .endObject().endObject());
+
+        DocumentMapper defaultMapper = createIndex("test2").mapperService().documentMapperParser()
+            .parse("type1", new CompressedXContent(falseMapping));
+        Mapper fieldMapper = defaultMapper.mappers().getMapper("location");
+        assertThat(fieldMapper, instanceOf(LegacyGeoShapeFieldMapper.class));
+
+        assertFalse(((LegacyGeoShapeFieldMapper)fieldMapper).fieldType().hasDocValues());
+
+        assertFieldWarnings("tree");
```

Comment:
Do we test proper defaults and somewhere?

Reply:
what do you mean?

Reply:
We have different defaults for doc_values for legacy and non-legacy shape. I wonder if we test these defaults.

Reply:
added tests for defaults